src/cryptography/hazmat/backends/openssl/ec.py

```diff
@@ -200,6 +200,23 @@ def private_numbers(self):
             public_numbers=self.public_key().public_numbers()
         )
 
+    def private_bytes(self, encoding, format, encryption_algorithm):
+        evp_pkey = self._backend._lib.EVP_PKEY_new()
+        assert evp_pkey != self._backend._ffi.NULL
+        evp_pkey = self._backend._ffi.gc(
+            evp_pkey, self._backend._lib.EVP_PKEY_free
+        )
+        res = self._backend._lib.EVP_PKEY_set1_EC_KEY(evp_pkey, self._ec_key)
```

Comment:
I don't think we have a choice about using EVP for PKCS8.

On Mon, Mar 2, 2015 at 7:39 AM, Alex Stapleton <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/ec.py
> <https://github.com/pyca/cryptography/pull/1709#discussion_r25607137>:
>
> > @@ -200,6 +200,23 @@ def private_numbers(self):
> >              public_numbers=self.public_key().public_numbers()
> >          )
> >
> > +    def private_bytes(self, encoding, format, encryption_algorithm):
> > +        evp_pkey = self._backend._lib.EVP_PKEY_new()
> > +        assert evp_pkey != self._backend._ffi.NULL
> > +        evp_pkey = self._backend._ffi.gc(
> > +            evp_pkey, self._backend._lib.EVP_PKEY_free
> > +        )
> > +        res = self._backend._lib.EVP_PKEY_set1_EC_KEY(evp_pkey, self._ec_key)
>
> I found that the EVP APIs get in the way and are often less obvious than
> the low level APIs for the kind of stuff we do but maybe I was
> insufficiently good at OpenSSL back then.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1709/files#r25607137>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084